src/factory/src/lib.rs

```diff
@@ -1,6 +1,6 @@
 pub mod api;
 mod error;
-mod state;
+pub mod state;
 
 pub use self::api::*;
```

Comment:
Not really related to this PR but I've been gone for a while so have to start somewhere:
It doesn't make sense to have a public module called `api` and then glob import everything there.

Remove the api module and put everything in this file.